src/sparse_ir/basis.py

```diff
@@ -225,6 +228,9 @@ def __init__(self, statistics, beta, wmax, eps=None, *, kernel=None,
         wmax = self.kernel.lambda_ / self.beta
         self.u = u.__class__(u.data, beta/2 * (u.knots + 1))
         self.v = v.__class__(v.data, wmax * v.knots)
+        roots_ = v[-1].roots()
+        self.sampling_points_v = np.hstack(
+            (-wmax, 0.5 * (roots_[0:-1] + roots_[1:]), wmax))
```

Comment:
See above.  This should be turned into a funciton and named consistently with the other sampling points ...